### ISSUE TITLE
formatter: non idempotent formatting on comments in TS

### ISSUE THREAD

**Issue by CatChen** [oxc-project/oxc#20219]
# Steps to reproduce

1. Clone https://github.com/CatChen/oxlint-suggestion-action and set it up with yarn v4.
2. Run `yarn codegen`. It calls `codegen.ts`, which has `afterOneFileWrite: ['oxfmt --write']` hook set up so all newly generated code should go through `oxfmt` once.
3. Commit any changes.
4. Run `yarn format`. It run `oxfmt` a second time.
5. Notice code change in `src/__graphql__/graphql.ts`.

Examples of code changes from the second pass of `oxfmt`:

```
@@ -1151,7 +1151,7 @@ export type AuditLogOrder = {
 1151
 1152    /** Properties by which Audit Log connections can be ordered. */
 1153    export type AuditLogOrderField =
 1154  - /** Order audit log entries by timestamp */
 1154  +   /** Order audit log entries by timestamp */
 1155      'CREATED_AT';
 1156
 1157    /** Represents a 'auto_merge_disabled' event on a given pull request. */
@@ -6010,7 +6010,7 @@ export type DeploymentOrder = {
 6010
 6011    /** Properties by which deployment connections can be ordered. */
 6012    export type DeploymentOrderField =
 6013  - /** Order collection by creation time */
 6013  +   /** Order collection by creation time */
 6014      'CREATED_AT';
 6015
 6016    /** A protection rule. */
@@ -7225,7 +7225,7 @@ export type EnterpriseAdministratorInvitationOrder = {
 7225
 7226    /** Properties by which enterprise administrator invitation connections can be ordered. */
 7227    export type EnterpriseAdministratorInvitationOrderField =
 7228  - /** Order enterprise administrator member invitations by creation time */
 7228  +   /** Order enterprise administrator member invitations by creation time */
 7229      'CREATED_AT';
 7230
 7231    /** The possible administrator roles in an enterprise account. */
@@ -7485,7 +7485,7 @@ export type EnterpriseMemberInvitationOrder = {
 7485
 7486    /** Properties by which enterprise member invitation connections can be ordered. */
 7487    export type EnterpriseMemberInvitationOrderField =
 7488  - /** Order enterprise member invitations by creation time */
 7488  +   /** Order enterprise member invitations by creation time */
 7489      'CREATED_AT';
 7490
 7491    /** Ordering options for enterprise member connections. */
```

They are purely indentations.

**Comment by leaysgur** [comment]
OK, I've minimized the root cause.

```ts
// ORIGINAL
export type AuditLogOrderField =
  /** Order audit log entries by timestamp */
  | 'CREATED_AT';

// First format
export type AuditLogOrderField =
/** Order audit log entries by timestamp */
  "CREATED_AT";

// Second format
export type AuditLogOrderField =
  /** Order audit log entries by timestamp */
  "CREATED_AT";
```

https://playground.oxc.rs/?t=formatter&formatterPanels=output%2Cprettier&options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Atrue%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22tsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Atrue%2C%22jsdoc%22%3Atrue%2C%22annotation%22%3Atrue%2C%22legal%22%3Atrue%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=%2F%2F+ORIGINAL%0Aexport+type+AuditLogOrderField+%3D%0A++%2F**+Order+audit+log+entries+by+timestamp+*%2F%0A++%7C+%27CREATED_AT%27%3B%0A%0A%2F%2F+First+format%0Aexport+type+AuditLogOrderField+%3D%0A%2F**+Order+audit+log+entries+by+timestamp+*%2F%0A++%22CREATED_AT%22%3B%0A%0A%2F%2F+Second+format%0Aexport+type+AuditLogOrderField+%3D%0A++%2F**+Order+audit+log+entries+by+timestamp+*%2F%0A++%22CREATED_AT%22%3B

**Comment by CatChen** [comment]
When there are multiple comment blocks it will move the first block to the previous line and remove the union value's indent in the first pass:

```
// ORIGINAL
export type AuditLogOrderField =
  /** Comment 1 */
  /** Comment 2 */
  /** Comment 3 */
  | 'CREATED_AT';

// First format
export type AuditLogOrderField = /** Comment 1 */
/** Comment 2 */
/** Comment 3 */
"CREATED_AT";

// Second format
export type AuditLogOrderField =
  /** Comment 1 */
  /** Comment 2 */
  /** Comment 3 */
  "CREATED_AT";
```

https://playground.oxc.rs/?t=formatter&formatterPanels=output%2Cprettier&options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Atrue%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22tsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Atrue%2C%22jsdoc%22%3Atrue%2C%22annotation%22%3Atrue%2C%22legal%22%3Atrue%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=%2F%2F+ORIGINAL%0Aexport+type+AuditLogOrderField+%3D%0A++%2F**+Comment+1+*%2F%0A++%2F**+Comment+2+*%2F%0A++%2F**+Comment+3+*%2F%0A++%7C+%27CREATED_AT%27%3B%0A%0A%2F%2F+First+format%0Aexport+type+AuditLogOrderField+%3D+%2F**+Comment+1+*%2F%0A%2F**+Comment+2+*%2F%0A%2F**+Comment+3+*%2F%0A%22CREATED_AT%22%3B%0A%0A%2F%2F+Second+format%0Aexport+type+AuditLogOrderField+%3D%0A++%2F**+Comment+1+*%2F%0A++%2F**+Comment+2+*%2F%0A++%2F**+Comment+3+*%2F%0A++%22CREATED_AT%22%3B

**Comment by CatChen** [comment]
Same issue for multiline comment block:

```
// ORIGINAL
export type AuditLogOrderField =
  /**
   * Comment
   */
  | 'CREATED_AT';

// First format
export type AuditLogOrderField =
/**
 * Comment
 */
  "CREATED_AT";

// Second format
export type AuditLogOrderField =
  /**
   * Comment
   */
  "CREATED_AT";
```

Behavior for multiple inline comments is broken as well:

```
// ORIGINAL
export type AuditLogOrderField =
  // Comment 1
  // Comment 2
  | 'CREATED_AT';

// First format
export type AuditLogOrderField = // Comment 1
// Comment 2
"CREATED_AT";

// Second format
export type AuditLogOrderField = // Comment 1
  // Comment 2
  "CREATED_AT";
```

**Comment by connorshea** [comment]
heh, I think I ran into the exact same problem with my app! Thanks for fixing :)
